ledger/src/blockstore_processor.rs

```diff
@@ -3613,8 +3613,9 @@ pub mod tests {
                     .unwrap()
                     .last_blockhash
             );
-            // ... but should affect bank hash
-            assert_ne!(ok_bank_details, bank_details);
+            // AND should not affect bankhash IF the rent is collected during freeze
+            // for the mint account (it is in this case)
+            assert_eq!(ok_bank_details, bank_details);
```

Comment:
Fee-payer is always charged. But the rent_epoch is only changed on success.
31/32 times the fee-payer would **not** be in the partition, it's rent epoch goes unchanged. That means that most of the time the hashes would not be equal.

Reply:
I was going off the name of the test - `test_transaction_result_does_not_affect_bankhash`. This to me implies that result should not affect bank hash.

However, yeah with this caveat carved out for rent collection stuff. We should test both paths - I will add a separate near identical test that transfers to a keypair NOT in first partition so that they will be different, and re-run all cases asserting inequality.

Reply:
[92659fd](https://github.com/anza-xyz/agave/pull/3916/commits/92659fd8a354ac6b2aa07beaf5023755ede48e60)